evap/results/tests/test_views.py

```diff
@@ -152,6 +153,58 @@ def test_wrong_state(self):
         url = '/results/semester/%s/course/%s' % (self.semester.id, course.id)
         self.get_assert_403(url, 'student')
 
+
+class TestResultsSemesterCourseDetailViewFewVoters(ViewTest):
+    @classmethod
+    def setUpTestData(cls):
+        semester = mommy.make(Semester, id=2)
+        mommy.make(UserProfile, username='evap', groups=[Group.objects.get(name='Staff')], email="evap@institution.example.com")
+        responsible = mommy.make(UserProfile, username='responsible')
+        student1 = mommy.make(UserProfile, username='student')
+        student2 = mommy.make(UserProfile)
+
+        course = mommy.make(Course, id=22, state='published', semester=semester, participants=[student1, student2], voters=[student1, student2])
+        questionnaire = mommy.make(Questionnaire)
+        cls.question_grade = mommy.make(Question, questionnaire=questionnaire, type="G")
+        question_likert = mommy.make(Question, questionnaire=questionnaire, type="L")
+        general_contribution = mommy.make(Contribution, contributor=None, course=course, questionnaires=[questionnaire])
```

Comment:
Shouldn't there already be a general_contribution stored in the course, since course.save() creates one if it's missing? Could you try if `course.general_contribution.questionnaires.set([questionnaire])` also does the job here?